0000-node-requirements/0000-node-requirements.md

```diff
@@ -0,0 +1,150 @@
+
+To correctly route responses, an ILP node must ensure it persists the state of an ILP Prepare packet, at least as long as specified in the **expiresAt** header. When a response to the ILP Prepare is received (either an ILP Fulfill or ILP Reject) from the outgoing link it MUST be routed down the same link on which the ILP Prepare was received. 
+
+If the packet expires before a response is received from the outgoing link then the connecter MUST send an ILP Reject packet as the response on the incoming link. Response packets that are received after the request has expired can be discarded.
+
+The link protocol used to communicate with peers MUST allow the node to match requests and responses such that when a response (ILP Fulfill or ILP Reject) is received from an outgoing link it can be matched to the original.
+
+## ILP Addresses
+
+An ILP packet is addressed using an ILP Address. The ILP Addressing scheme is similar to the IP addressing scheme (both IPv4 and IPv6) in that the scheme is a hierarchical address space where routing is done based on (primarily) longest prefix match of a destination address against the address range prefixes in the node's routing table.
+
+Unlike IP addresses, ILP addresses are variable in size and encoded as IA7 (ASCII) strings of dot (`.`) separated segments. The valid characters in an address segment are alphanumeric plus Underscore (`_`), Tilde (`~`) and Hyphen (`-`) and are case sensitive which allows base64 encoded data to be encoded into address segments.
+
+More details on ILP Addresses are in [ILP RFC 15](https://interledger.org/rfcs/0015-ilp-addresses/).
+
+## Amounts
+
+Amounts in ILP are simple and are represented as 64-bit unsigned integers. As such an amount will have an associated currency and scale which specifies how the integer representation can be converted to a more user friendly decimal form when required.
+
+For example a link between two peers on the network may be settled in US dollars and the peers may decide they wish to allow precision up to 100ths of a cent. In this case the link will be configured by both peers to use the currency USD and the scale of 4. An ILP packet with an amount of 12345 is $1.23,45 or one dollar and twenty-three point four five cents.
+
+# Requirements
+
+## Links
+
+A node must have one or more links to peer nodes with whom it will exchange ILP packets. These links can use any protocol to exchange the packets as long as both peers are able to **correctly correlate which packets are responses to a specific previous request**.
+
+When two peers establish a link they must **agree on the settlement currency and scale for packets exchanged over that link**. The amount in an ILP packet is an unsigned 64-bit number, therefor when a node receives a packet it must infer the currency and scale of the amount from the link on which it was received.
```

Comment:
therefor -> therefore

Reply:
hrmmm, am I confused ? 

https://www.grammarly.com/blog/therefore-vs-therefor/

Reply:
No, just a bad habit of mine